app/services/authentication/providers/provider.rb

```diff
@@ -22,8 +22,7 @@ def existing_user_data
       end
 
       def user_nickname
-        # Apple doesn't have the `nickname` field in the `info` payload
-        (info.first_name&.downcase || info.nickname).to_s
+        info.nickname.to_s
```

Comment:
isn't `info.nickname` empty for Apple?

Reply:
Yes, but we can use `info.nickname` for all providers (keeping it in the base class `Provider`) and whenever we implement a new provider that is different (Apple 😆) we can override it in their own class `Authentication::Providers::Apple::user_nickname`.

Hypothetical example: If we add Linkedin sign in and their hash __does have__ `info.nickname` then no need to override. Later on we implement Facebook auth and they __don't have__ `info.nickname` but instead they have something like `info.profile_name` we can just override it in `Authentication::Providers::Facebook::user_nickname`.

This way no matter which provider we're referring to they will have `user_nickname` respond with a valid username and the base class implementation (most common scenario) remains a bit cleaner/simpler.

An alternative I thought about was making it a "virtual method" (using `raise SubclassResponsibility`) so each provider can implement their own. Maybe this sounds better?

Reply:
Makes sense, thank you!